pkgs/development/python-modules/pyarrow/default.nix

```diff
@@ -65,10 +65,23 @@ buildPythonPackage rec {
   PYARROW_WITH_PARQUET = zero_or_one true;
   PYARROW_WITH_PLASMA = zero_or_one (!stdenv.isDarwin);
   PYARROW_WITH_S3 = zero_or_one _arrow-cpp.enableS3;
-
-  PYARROW_CMAKE_OPTIONS = [
-    "-DCMAKE_INSTALL_RPATH=${ARROW_HOME}/lib"
-  ];
+  # do this manually to avoid having to patch pyarrow, to handle pyarrow's
+  # assumption of permissions of bundled arrow-cpp headers
+  PYARROW_BUNDLE_ARROW_CPP_HEADERS = zero_or_one false;
+
+  PYARROW_CMAKE_OPTIONS =
+    let
+      components = [
+        "Arrow"
+        "ArrowDataset"
+        "ArrowFlight"
+        "Parquet"
+        "Plasma"
```

Comment:
Since plasma has been [deprecated](https://arrow.apache.org/docs/python/plasma.html) we might just remove it now?

Reply:
Yep, will push that up in a bit.

Reply:
For those curious about deprecation, here is a link https://lists.apache.org/thread/nw232k2lzmg9kcl8ts475m9ybl34j81p